src/mesido/electricity_physics_mixin.py

```diff
@@ -640,8 +681,19 @@ def __electrolyzer_path_constaint(self, ensemble_member):
             elif (
                 options["electrolyzer_efficiency"]
                 == ElectrolyzerOption.LINEARIZED_THREE_LINES_WEAK_INEQUALITY
+                or options["electrolyzer_efficiency"]
+                == ElectrolyzerOption.LINEARIZED_THREE_LINES_EQUALITY
             ):
-                curve_fit_number_of_lines = 3
+                if (
+                    options["electrolyzer_efficiency"]
+                    == ElectrolyzerOption.LINEARIZED_THREE_LINES_WEAK_INEQUALITY
+                ):
+                    curve_fit_number_of_lines = 3
+                else:
+                    curve_fit_number_of_lines = len(
+                        self.__electrolyzer_is_active_linear_segment_map
+                    )
+
```

Comment:
curve_fit_number_of_lines should always be three for LINEARIZED_THREE_LINES_EQUALITY and LINEARIZED_THREE_LINES_WEAK_INEQUALITY as the option names state. Otherwise we'll change the options and we'll provide it as a general setting parameter.